sys/Makefile.include

```diff
@@ -107,6 +107,9 @@ ifneq (,$(filter skald, $(USEMODULE)))
 endif
 
 ifneq (,$(filter sock sock_%,$(USEMODULE)))
+  ifneq (,$(filter ipv4,$(USEMODULE)))
+    CFLAGS += -DSOCK_HAS_IPV4
+  endif
```

Comment:
This should be its own PR IMHO, so this behavioral change to a very central API is better tracable in the future.

Reply:
Does this also include 47abb8e?

Reply:
#17511